baseplate_test.go

```diff
@@ -149,6 +150,72 @@ func TestServe(t *testing.T) {
 	}
 }
 
+type timestampCloser struct {
+	ts []time.Time
+}
+
+func (c *timestampCloser) Close() error {
+	c.ts = append(c.ts, time.Now())
+	return nil
+}
+
+func TestServeClosers(t *testing.T) {
+	t.Parallel()
+
+	store := newSecretsStore(t)
+	defer store.Close()
+
+	bp := baseplate.NewTestBaseplate(baseplate.Config{StopTimeout: testTimeout}, store)
+
+	pre := &timestampCloser{}
+	post := &timestampCloser{}
+
+	args := baseplate.ServeArgs{
+		Server:       newWaitServer(t, bp, time.Millisecond),
+		PreShutdown:  []io.Closer{pre},
+		PostShutdown: []io.Closer{post},
+	}
+
+	t.Run(
+		"serve closer order",
+		func(t *testing.T) {
+			ch := make(chan error)
+
+			go func() {
+				// Run Serve in a goroutine since it is blocking
+				ch <- baseplate.Serve(
+					context.Background(),
+					args,
+				)
+			}()
+
+			time.Sleep(time.Millisecond)
+			p, err := os.FindProcess(syscall.Getpid())
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			p.Signal(os.Interrupt)
+			err = <-ch
```

Comment:
I cribbed this from `TestServe`, but am not actually too sure if this block (and the provided `args.Server`) are correct for testing a happy shutdown path.

Reply:
Seems fine to me, do you have any specific concerns?

Reply:
`err` is assigned but never used in its current state? Do you mean to check it? if not you can just do `<-ch` without assigning it to anything.

Reply:
🔕 We probably should check `err != nil` though?